translator/gobgp.py

```diff
@@ -83,20 +83,29 @@ def _build_path(self, ip, event_data={}):
         as_path.Pack(as_segments)
 
         # Set our community number
-        # global_admin = The Global Administrator field is intended to represent a complete 4-byte ASN
-        # local_data* is for more flexible routing policy. can be set to ME:ACTION:YOU or ASN:Function:Parameter.
-        #
-        # Use local_data2 for any "flag" you want to send to the router that can be used for routing policy decisions.
+        # The ASN gets packed into the community so we need to be careful about size to not overflow the structure
         communities = Any()
-        global_admin = asn
-        local_data1 = community
-        local_data2 = 0
-        large_community = attribute_pb2.LargeCommunity(
-            global_admin=global_admin,
-            local_data1=local_data1,
-            local_data2=local_data2,
-        )
-        communities.Pack(attribute_pb2.LargeCommunitiesAttribute(communities=[large_community]))
+        # Standard community
+        # Since we pack both into the community string we need to make sure they will both fit
+        if asn < 65536 and community < 65536:
+            comm_id = (asn << 16) + community
+            communities.Pack(attribute_pb2.CommunitiesAttribute(communities=[comm_id]))
+        else:
+            # Large Community: 3 fields of 4 bytes version of a community string
+            # global_admin = The Global Administrator field is intended to represent a complete 4-byte ASN
+            # local_data* is for more flexible routing policy. can be set to ME:ACTION:YOU or ASN:Function:Parameter.
+            #
+            # Use local_data2 for any "flag" you want to send to the router to be used for routing policy decisions.
+            logging.warning(f"LargeCommunity Used - ASN:{asn} Community: {community}")
```

Comment:
I don't think this should be a warning. If your peering supports it, it supports it.

Reply:
Fair point, logging.info should suffice. I just wanted to make sure people had some way to figure out why their routes were getting discarded if they accidentally typoed the ASN and ended up using a LargeCommunity by mistake.